lib/CodeGen/CGExpr.cpp

```diff
@@ -2046,6 +2047,115 @@ void CodeGenFunction::emitAsmReturnDataCopy(const CallExpr *CE) {
   CGM.emitReturnDataCopy(Pos, Ptr, Length);
 }
 
+llvm::Value *CodeGenFunction::emitAsmCall(const CallExpr *CE) {
+  auto Arguments = CE->getArguments();
+
+  llvm::Value *Gas = emitExpr(Arguments[0])->load(Builder, CGM);
```

Comment:
Looks like you assumed Arguments is not empty. Should we add an assertion to check the size of Arguments before we access it directly?

Reply:
For example, like #49 which will show the error message to tell users that the given arguments are not enough for this function call.